skbio/diversity/_base.py

```diff
@@ -0,0 +1,85 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import numpy as np
+
+from skbio.tree import DuplicateNodeError, MissingNodeError
+
+
+def _validate_counts_vector(counts, suppress_cast=False):
+    """Validate and convert input to an acceptable counts vector type.
+
+    Note: may not always return a copy of `counts`!
+
+    """
+    counts = np.asarray(counts)
+
+    if not suppress_cast:
+        counts = counts.astype(int, casting='safe', copy=False)
+
+    if counts.ndim != 1:
+        raise ValueError("Only 1-D vectors are supported.")
+    elif (counts < 0).any():
+        raise ValueError("Counts vector cannot contain negative values.")
+
+    return counts
+
+
+def _validate_counts_vectors(u_counts, v_counts, suppress_cast=False):
```

Comment:
Comment from @jairideout: maybe generalize this to take *args, so it can work for one or more counts vectors. This will allow consolidation with ``_validate_counts_vector``.